spec/services/more_like_this_getter_spec.rb

```diff
@@ -0,0 +1,14 @@
+require 'rails_helper'
+# Testing the SOLR part of this is slow -- and a pain --
+# so we are short-circuiting that part of the code.
+describe MoreLikeThisGetter do
+  let(:work_to_match)   { create(:public_work)  }
+  let(:getter) {MoreLikeThisGetter.new(work_to_match) }
+  let(:three_public_work_ids) { [1,2,3].map {create(:public_work).friendlier_id } }
+  let(:matching_work_ids) { three_public_work_ids + [create(:private_work).friendlier_id]} 
+
+  it "returns only published works; displays them in order" do
+    allow(getter).to receive(:friendlier_ids).and_return(matching_work_ids)
+    expect(getter.works.map {|w| w.friendlier_id}).to eq (matching_work_ids[0..2])
```

Comment:
looking around at the code, I see that this second line is meant to test that it only returned published works, not private ones. 

That is kind of hard to see from the code though. Is there a way to make it more apparent what the test is testing? The magic `matching_work_ids[0..2]` is cryptic; could it instead just be testing against a `public_works` variable or something?

It's also not clear to me what the "order" is -- is it the order that Solr returns it in, like most "similar" first?  How does this test test that?  Could it be made more clear in code how it's testing that, and/or comments, _what_ "order" is expected, in order of what?

The comments say "Testing the SOLR part of this is slow -- and a pain -- so we are short-circuiting that part of the code" -- I'm not sure what this means. This code isn't _really_ going to solr?

Hmm, can we find a way to actually go to solr?  What makes it a pain, that we aren't sure what solr will return as "more like this"?

I think it would be better if we had just one test at least actually going to solr.

Reply:
I was able to figure out how to get SOLR to give us more-like-this results in the test environment, so we do have this now.